src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs

```diff
@@ -566,10 +555,11 @@ public virtual void EndWriteAttributeTo(TextWriter writer)
 
             if (value != null)
             {
+                // Perf: We'll use this buffer for all of the attribute values and then clear it to
+                // reduce allocations.
                 if (_valueBuffer == null)
                 {
-                    var buffer = new ViewBuffer(BufferScope, Path);
-                    _valueBuffer = new HtmlContentWrapperTextWriter(buffer, Output.Encoding);
```

Comment:
Creating a buffer for this small of content was not good. We just use one stringwriter per page now.